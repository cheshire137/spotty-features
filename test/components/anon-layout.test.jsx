import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import AnonLayout from '../../src/components/anon-layout.jsx'

describe('AnonLayout', () => {
  let component = null

  beforeEach(() => {
    component = <AnonLayout children={<div>hey</div>} />
  })

  test('matches snapshot', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('has title', () => {
    const title = shallow(component).find('.title a')
    expect(title.text()).toBe('Spotty Features')
  })

  test('includes children', () => {
    const content = shallow(component).find('.content-container')
    expect(content.children().length).toBe(1)
    expect(content.children().text()).toBe('hey')
  })
})
