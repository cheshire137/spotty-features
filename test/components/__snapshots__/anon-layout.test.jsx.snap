exports[`AnonLayout matches snapshot 1`] = `
<div>
  <div
    className="hero is-primary">
    <div
      className="hero-body">
      <div
        className="container">
        <h1
          className="title">
          <a
            href="/">
            Spotty Features
          </a>
        </h1>
      </div>
    </div>
  </div>
  <section
    className="section">
    <div
      className="content-container container">
      <div>
        hey
      </div>
    </div>
  </section>
</div>
`;
